terraform/eks-cloudwatch/app_mesh.tf

```diff
@@ -0,0 +1,156 @@
+# ------------------------------------------------------------------------
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+# creating a eks cluster takes around 10 minutes typically.
+# so in the eks/k8s test, we need tester to provide the cluster instead of creating it in terraform
+# so that we can shorten the execution time
+
+locals {
+  aws_account_id = var.aws_account_id != "" ? var.aws_account_id : data.aws_caller_identity.current.account_id
+  mesh_name = "howto-k8s-http-headers"
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy" "appmesh_policy" {
+  arn = "arn:aws:iam::${local.aws_account_id}:policy/AWSAppMeshK8sControllerIAMPolicy"
+}
+
+module "iam_assumable_role_with_oidc" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version = "~> 3.0"
+  provider_url = data.aws_eks_cluster.testing_cluster.identity[0].oidc[0].issuer
+
+  create_role = true
+  role_name = "appmesh-controller_role"
+  role_policy_arns = [
+    data.aws_iam_policy.appmesh_policy.arn]
+
+  oidc_fully_qualified_subjects = [
+    "system:serviceaccount:${kubernetes_namespace.appmesh_ns.metadata[0].name}:appmesh-controller"]
+  number_of_role_policy_arns = 1
+}
+
+resource "kubernetes_namespace" "appmesh_ns" {
+  metadata {
+    name = "appmesh-system-${module.common.testing_id}"
+  }
+}
+
+resource "kubernetes_service_account" "appmesh_sa" {
+  metadata {
+    name = "appmesh-controller"
+    namespace = kubernetes_namespace.appmesh_ns.metadata[0].name
+    annotations = {
+      "eks.amazonaws.com/role-arn" = module.iam_assumable_role_with_oidc.this_iam_role_arn
+    }
+  }
+  automount_service_account_token = true
+}
+
+provider "helm" {
+  kubernetes {
+    config_path = "~/.kube/config"
+  }
+}
+
+resource "helm_release" "eks" {
+  name = "eks"
+  namespace = kubernetes_namespace.appmesh_ns.metadata[0].name
+
+  repository = "https://aws.github.io/eks-charts"
+  chart = "appmesh-controller"
+  version = "1.3.0"
```

Comment:
why 1.3.0?

Reply:
Fix the version of AppMesh to avoid version conflict